ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java

```diff
@@ -1163,9 +1163,15 @@ private static class MaterializationRebuildLockHeartbeater implements Runnable {
     public void run() {
       LOG.trace("Heartbeating materialization rebuild lock for {} for query: {}",
           AcidUtils.getFullTableName(dbName, tableName), queryId);
-      boolean refreshed;
+      boolean refreshed = false;
       try {
-        refreshed = txnMgr.heartbeatMaterializationRebuildLock(dbName, tableName, txnId);
+        if (!txnMgr.getValidTxns().isTxnAborted(txnId)) {
```

Comment:
IIUC only MV rebuild txns are counts here.
Can we use `getValidTxns(List<TxnType> excludeTxnTypes)` and exclude everything but `MATER_VIEW_REBUILD`?

Reply:
Changed as suggested.